hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDownloadArtifactResource.java

```diff
@@ -63,6 +64,9 @@
 
         final SoftwareModule module = softwareModuleManagement.get(softwareModuleId)
                 .orElseThrow(() -> new EntityNotFoundException(SoftwareModule.class, softwareModuleId));
+        if (module.isDeleted()) {
```

Comment:
We also should not expose the "download" link at all in case the Software Module is soft deleted (see MgmtSoftwareModuleResource#getArtifact -> MgmtSoftwareModuleMapper.addLinks)

Reply:
done